### PR TITLE
Return a schema together with the response data

### PR DESCRIPTION
**Pull request by lava**

### CODE REVIEW

File: libvast/builtins/endpoints/export.cpp [hunk 1]
```diff
@@ -202,17 +460,38 @@ export_multiplexer_actor::behavior_type export_multiplexer(
...
         // Should be type-checked by the server.
         VAST_ASSERT(caf::holds_alternative<count>(param));
-        limit = caf::get<count>(param);
+        params.limit = caf::get<count>(param);
+      }
+      if (rq.params.contains("flatten")) {
+        auto& param = rq.params.at("flatten");
+        VAST_ASSERT(caf::holds_alternative<bool>(param));
+        params.format_opts.flatten = caf::get<bool>(param);
+      }
```
Comment (tobim) [inline libvast/builtins/endpoints/export.cpp:481]: Isn't something a little more concise possible as well?
```suggestion
      params.format_opts.flatten = caf::get_or(rq.params, "flatten", params.format_opts.flatten);
```

File: libvast/CMakeLists.txt [hunk 2]
```diff
@@ -334,15 +334,8 @@ endif ()
...
-  file(STRINGS "${source}" lines)
-  foreach (line IN LISTS lines)
-    if ("${line}" MATCHES "^ *VAST_REGISTER_PLUGIN *\\(.+\\) *$")
-      set(has_register TRUE)
-      break()
-    endif ()
-  endforeach ()
-  if (NOT "${has_register}")
+  file(READ "${source}" lines)
+  if (NOT "${lines}" MATCHES "\n *VAST_REGISTER_PLUGIN *\\(.+\\) *[\n$]")
```
Comment (tobim) [inline libvast/CMakeLists.txt:338]: ???

File: libvast/builtins/endpoints/export.cpp [hunk 3]
```diff
@@ -89,17 +210,167 @@ using export_multiplexer_actor = system::typed_actor_fwd<>
...
+/// Format a set of table slices like this:
+//
+//  {
+//    "version": "v2.4-rc2",
+//    "num_events": 3,
+//    "events": [
+//      {
+//        "name": "zeek.conn",
+//        "schema": [{"name": "_path", "type": "string"}, {"name": "uid",
+//        "type": "string"}, ...] "data": [
```
Comment (tobim) [inline libvast/builtins/endpoints/export.cpp:282]: The formatting looks off here.

**Review by tobim — COMMENTED** [review]
I think this is a great start. Now we only need to agree whether to optimize the data in the type grouped case.

File: libvast/builtins/endpoints/export.cpp [hunk 4]
```diff
@@ -65,11 +89,108 @@ static auto const* SPEC_V0 = R"_(
...
+    description: Export data from VAST according to a query. The query must be a valid expression in the VAST Query Language. (see https://vast.io/docs/understand/query-language)
+    parameters:
+      - in: query
+        name: expression
+        schema:
+          type: string
+        required: true
+        default: A query matching every event.
+        description: The query expression to execute.
+        example: ":ip in 10.42.0.0/16"
```
[the remark this answers is not in the document]
Comment (tobim) [reply libvast/builtins/endpoints/export.cpp:114]: ... unless `ip` is a type alias for `addr`.

**Comment by lava** [comment]
I rebased this on top of #2773 and am closing this PR in favor of that one to keep the review/feedback overhead smaller.

**Pull request closed without merging**